pycbc/strain/strain.py

```diff
@@ -100,26 +99,27 @@ def detect_loud_glitches(strain, psd_duration=4., psd_stride=2.,
         logging.info('Autogating: downsampling strain')
         strain = resample_to_delta_t(strain, 0.5 / high_freq_cutoff,
                                      method='ldas')
+    else:
+        strain = strain.copy()
 
     logging.info('Autogating: tapering strain')
-    taper_length = int(corrupt_time * strain.sample_rate)
-    w = numpy.arange(taper_length) / float(taper_length)
-    strain[0:taper_length] *= pycbc.types.Array(w, dtype=strain.dtype)
-    strain[(len(strain)-taper_length):] *= pycbc.types.Array(w[::-1],
-                                                             dtype=strain.dtype)
+    corrupt_length = int(corrupt_time * strain.sample_rate)
+    w = numpy.arange(corrupt_length) / float(corrupt_length)
+    strain[0:corrupt_length] *= pycbc.types.Array(w, dtype=strain.dtype)
+    strain[(len(strain)-corrupt_length):] *= \
+            pycbc.types.Array(w[::-1], dtype=strain.dtype)
+
     if output_intermediates:
         strain.save_to_wav('strain_conditioned.wav')
 
-    corrupt_length = int(corrupt_time * strain.sample_rate)
-
     # zero-pad strain to a power-of-2 length
     strain_pad_length = next_power_of_2(len(strain))
     pad_start = int(strain_pad_length/2 - len(strain)/2)
     pad_end = pad_start + len(strain)
     strain_pad = pycbc.types.TimeSeries(
             pycbc.types.zeros(strain_pad_length, dtype=strain.dtype),
             delta_t=strain.delta_t, copy=False,
-            epoch=strain.start_time-pad_start/strain.sample_rate)
+            epoch=strain.start_time-pad_start/float(strain.sample_rate))
```

Comment:
PEP8 here (spaces around `-` `/`)

Reply:
Does that apply also when the operation is used as a kwarg?

Reply:
yes please!  (.. it would if you calculated the strain epoch on a separate line of code, which might actually be more readable)